x/incentive/keeper/rewards_test.go

```diff
@@ -0,0 +1,341 @@
+	suite.keeper.SetRewardPeriod(suite.ctx, rp)
+	suite.keeper.SetNextClaimPeriodID(suite.ctx, "bnb", 1)
+	suite.NotPanics(func() {
+		suite.keeper.HandleRewardPeriodExpiry(suite.ctx, rp)
+	})
+	_, found := suite.keeper.GetClaimPeriod(suite.ctx, 1, "bnb")
+	suite.True(found)
+
+	addr, _ := sdk.AccAddressFromBech32("kava15qdefkmwswysgg4qxgqpqr35k3m49pkx2jdfnw")
+	c1 := types.NewClaim(addr, c("ukava", 1000000), "bnb", 1)
+	suite.keeper.SetClaim(suite.ctx, c1)
+	suite.NotPanics(func() {
+		suite.keeper.AddToClaim(suite.ctx, addr, "bnb", 1, c("ukava", 1000000))
+	})
+	testC, _ := suite.keeper.GetClaim(suite.ctx, addr, "bnb", 1)
+	suite.Equal(c("ukava", 2000000), testC.Reward)
+
+	suite.NotPanics(func() {
+		suite.keeper.AddToClaim(suite.ctx, addr, "xpr", 1, c("ukava", 1000000))
+	})
+
+	suite.SetupTest()
+	reward := types.NewReward(true, "bnb", c("ukava", 1000000000), time.Hour*7*24, time.Hour*24*365, time.Hour*7*24)
+	suite.NotPanics(func() {
+		suite.keeper.CreateNewRewardPeriod(suite.ctx, reward)
+	})
+	_, found = suite.keeper.GetRewardPeriod(suite.ctx, "bnb")
+	suite.True(found)
+
+}
+
+func (suite *KeeperTestSuite) TestCreateAndDeleteRewardsPeriods() {
+	reward1 := types.NewReward(true, "bnb", c("ukava", 1000000000), time.Hour*7*24, time.Hour*24*365, time.Hour*7*24)
+	reward2 := types.NewReward(false, "xrp", c("ukava", 1000000000), time.Hour*7*24, time.Hour*24*365, time.Hour*7*24)
+	params := types.NewParams(true, types.Rewards{reward1, reward2})
+	suite.keeper.SetParams(suite.ctx, params)
+
+	suite.NotPanics(func() {
+		suite.keeper.CreateAndDeleteRewardPeriods(suite.ctx)
+	})
+
+	_, found := suite.keeper.GetRewardPeriod(suite.ctx, "bnb")
+	suite.True(found)
+	_, found = suite.keeper.GetRewardPeriod(suite.ctx, "xrp")
+	suite.False(found)
+
+}
+
+func (suite *KeeperTestSuite) TestApplyRewardsToCdps() {
+	suite.setupCdpChain()
+	suite.ctx = suite.ctx.WithBlockTime(suite.ctx.BlockTime().Add(time.Second * 100))
+	suite.NotPanics(func() {
+		suite.keeper.ApplyRewardsToCdps(suite.ctx)
+	})
+	claims := types.Claims{}
+	suite.keeper.IterateClaims(suite.ctx, func(c types.Claim) (stop bool) {
+		claims = append(claims, c)
+		return false
+	})
+	suite.Equal(3, len(claims))
+	_, found := suite.keeper.GetClaimPeriod(suite.ctx, 1, "bnb")
+	suite.False(found)
+
+	// move to the past the period expiry and check that the claim period has been created and the next claim period id has increased
+	suite.ctx = suite.ctx.WithBlockTime(suite.ctx.BlockTime().Add(time.Hour * 24 * 7))
+	suite.NotPanics(func() {
+		suite.keeper.ApplyRewardsToCdps(suite.ctx)
+	})
+	_, found = suite.keeper.GetClaimPeriod(suite.ctx, 1, "bnb")
+	suite.True(found)
+	testID := suite.keeper.GetNextClaimPeriodID(suite.ctx, "bnb")
+	suite.Equal(uint64(2), testID)
+}
+
+func (suite *KeeperTestSuite) addObjectsToStore() {
```

Comment:
Shared functionality that can be moved to keeper_test.go as it's called from both rewards_test.go and querier_test.go.